From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 27 Feb 2024 10:54:52 +0100
Subject: tests: Compare floats with an epsilon

Due to how floating point numbers are represented, there's typically
a tiny error margin. Compare doubles with a DBL_EPSILON wiggle room, so
we don't trip up with these such errors on strange architectures.

(cherry picked from commit 79e79f3b2fd4d69bec4a3796e650c8edbc5af33f)

Origin: future 3.7.rc
---
 tests/libtracker-sparql/tracker-batch-test.c     | 2 +-
 tests/libtracker-sparql/tracker-statement-test.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/libtracker-sparql/tracker-batch-test.c b/tests/libtracker-sparql/tracker-batch-test.c
index c35e66e..e0011e3 100644
--- a/tests/libtracker-sparql/tracker-batch-test.c
+++ b/tests/libtracker-sparql/tracker-batch-test.c
@@ -250,7 +250,7 @@ assert_photo (TestFixture *test_fixture,
 
 	g_assert_true (tracker_sparql_cursor_next (cursor, NULL, &error));
 	g_assert_cmpstr (tracker_sparql_cursor_get_string (cursor, 0, NULL), ==, TRACKER_PREFIX_NMM "exposureTime");
-	g_assert_cmpfloat (tracker_sparql_cursor_get_double (cursor, 1), ==, exposure_time);
+	g_assert_cmpfloat_with_epsilon (tracker_sparql_cursor_get_double (cursor, 1), exposure_time, DBL_EPSILON);
 
 	/* Skip over nrl:added, nrl:modified, rdf:type */
 
diff --git a/tests/libtracker-sparql/tracker-statement-test.c b/tests/libtracker-sparql/tracker-statement-test.c
index 33d9a0f..356b2e1 100644
--- a/tests/libtracker-sparql/tracker-statement-test.c
+++ b/tests/libtracker-sparql/tracker-statement-test.c
@@ -290,7 +290,7 @@ rdf_types (TestFixture   *test_fixture,
 	g_assert_cmpstr (tracker_sparql_cursor_get_string (cursor, 0, &len), ==, "Hello");
 	g_assert_cmpint (len, ==, strlen ("Hello"));
 	g_assert_cmpint (tracker_sparql_cursor_get_integer (cursor, 1), ==, 42);
-	g_assert_cmpfloat (tracker_sparql_cursor_get_double (cursor, 2), ==, 42.2);
+	g_assert_cmpfloat_with_epsilon (tracker_sparql_cursor_get_double (cursor, 2), 42.2, DBL_EPSILON);
 	g_assert_cmpint (tracker_sparql_cursor_get_boolean (cursor, 3), ==, TRUE);
 	g_assert_true (g_date_time_equal (tracker_sparql_cursor_get_datetime (cursor, 4), datetime));
 	g_assert_cmpstr (tracker_sparql_cursor_get_langstring (cursor, 5, &langtag, &len), ==, "Hola");
